plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/TestGeneratorStatechartGenerator.xtend

```diff
@@ -60,14 +60,14 @@ class TestGeneratorStatechartGenerator extends AbstractContractStatechartGenerat
 	val boolean coldViolationExisits
 	val StatechartGenerationMode generationMode
 
-	new(ScenarioDefinition scenario, Component component, StatechartGenerationMode mode,
+	new(ScenarioDeclaration scenario, Component component, StatechartGenerationMode mode,
 		boolean dedicatedColdViolation) {
 		super(scenario, component)
 		this.generationMode = mode
 		this.coldViolationExisits = dedicatedColdViolation
 	}
 
-	new(ScenarioDefinition scenario, Component component) {
+	new(ScenarioDeclaration scenario, Component component) {
```

Comment:
Similarly... Can the `component` not be acquired here inside the constructor using a derived feature? If so, it should not be added here as a separate parameter.